compiler/modules/front.syntax/ast.bal

```diff
@@ -465,12 +469,19 @@ public type MappingTypeDesc record {|
     t:MappingDefinition? defn = ();
 |};
 
+public type FunctionTypeParam record {|
```

Comment:
@jclark I removed the sub-typing here because it seems to cause an intermittent java null ptr exception in the compiler 
```
[2021-11-17 05:19:54,592] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isSameType(Types.java:341)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types$BSameTypeVisitor.visit(Types.java:2542)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types$BSameTypeVisitor.visit(Types.java:2417)
	at org.wso2.ballerinalang.compiler.semantics.model.types.BRecordType.accept(BRecordType.java:75)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isSameType(Types.java:341)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isSameType(Types.java:306)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isAssignable(Types.java:638)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isAssignableToUnionType(Types.java:3157)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isAssignable(Types.java:744)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isArrayTypesAssignable(Types.java:1175)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isAssignable(Types.java:811)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isAssignable(Types.java:633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.lambda$resolveIncludedFields$32(SymbolEnter.java:4618)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.resolveIncludedFields(SymbolEnter.java:4630)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.resolveFields(SymbolEnter.java:3741)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineFieldsOfObjectOrRecordTypeDef(SymbolEnter.java:3679)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineFields(SymbolEnter.java:3633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineConstructs(SymbolEnter.java:404)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.visit(SymbolEnter.java:318)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:163)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.defineNode(SymbolEnter.java:275)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter.definePackage(SymbolEnter.java:268)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:397)
	at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:349)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:201)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:186)
	at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:111)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:100)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
	at io.ballerina.projects.Package.getCompilation(Package.java:140)
	at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:75)
	at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
	at io.ballerina.cli.cmd.TestCommand.execute(TestCommand.java:211)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.ballerina.cli.launcher.Main.main(Main.java:51)
```

Reply:
@rdhananjaya Can you please create report an issue or help Heshan to do so.